### DISCUSSION TITLE
Embed mp4 video

### DISCUSSION THREAD

**Discussion by Sciss** [mmistakes/minimal-mistakes#101] [Q&A]
I like the simplicity of `{% audio myfile.mp3 %}`. Is there an equivalent for videos?

I tried `{% video myfile.mp4 %}`, that doesn't work. Also HTML directly has a problem:

```
<video width="480" height="320" controls="controls">
  <source src="myfile.mp4" type="video/mp4">
</video>
```

The latter produces a Jekyll error "Liquid exception: Unknown tag 'video'"

What is the best approach to embed videos (not hosted on Youtube, Vimeo or such!).

**Comment by Sciss** [comment]
Ok, I think I found the solution. One can copy the [Octopress Video Tag plugin](http://octopress.org/docs/plugins/video-tag/) into the `_plugins` folder, then it's possible to use `{% video mfile.mp4 width height mythumb.jpg %}`.

**Comment by mmistakes** [comment]
I've never used those Jekyll filters to do audio or video. I always embed videos with straight HTML in my MarkDown files. If you remove the two tags and leave just the HTML it will work. Liquid is barfing on your `{% video %}` tag because it doesn't know what it is.

Here's what my `2014-11-18-mp4-video-test.md` looks like.

``` html
---
layout: post
title: "MP4 Video"
modified:
categories: 
excerpt:
tags: []
image:
  feature:
date: 2014-11-18T11:34:04-05:00
---

This is testing out Liquid.

<video width="480" height="320" controls="controls">
  <source src="myfile.mp4" type="video/mp4">
</video>
```

And this is how it compiled with Jekyll.
![mp4-video](https://cloud.githubusercontent.com/assets/1376749/5091858/7a3467d0-6f19-11e4-9a5f-3678503417c2.jpg)

That Octopress plugin is a good solution if you don't host on GitHub. I don't think that plugin is whitelisted and it won't work if that's the case.

**Comment by Sciss** [comment]
> If you remove the two tags and leave just the HTML it will work. Liquid is barfing on your {% video %} tag because it doesn't know what it is.

Ah indeed, somehow I got it mixed up, I thought the problem was `<video/>`, perhaps I forgot to remove a `{% video ... %}` and didn't notice. Nice.

> That Octopress plugin is a good solution if you don't host on GitHub. 

Yes, I'm hosting else-where and anyway I'm just pushing the rendered HTML, so the plugin is transparent.

**Comment by sonichn** [comment]
Hi,

I am using GitHub flavored markdown.  Unable to embed .mp4 video in markdown.

I am using the code:

* {
<video  style="width:100%" controls>
  <source src="sample.mp4" type="video/mp4">
Your browser does not support the video tag.
</video>
}*

still I'm not able to view the video.
Are any files required to be installed.

**Comment by beans816** [comment]
https://github.blog/2021-05-13-video-uploads-available-github/

(saw this thread when looking for the link)

it has huge limitations but it is possible now

sadly github doesnt support `<video>` or `<style>` quite yet
